src/_nebari/keycloak.py

```diff
@@ -29,7 +29,10 @@ def do_keycloak(config: schema.Main, *args):
 
         username = args[1]
         password = args[2] if len(args) >= 3 else None
-        create_user(keycloak_admin, username, password, domain=config.domain)
+        groups = args[3] if len(args) >= 4 else None
```

Comment:
I'm not a fan of this approach to passing arguments around but did this to follow the pattern already defined in this part of the code. 
I would be happy to refactor this module to make it a more developer friendly if that's something the nebari team is into.

Reply:
why don't you split that into a different pr. Thanks!